### PR TITLE
fix: update zip.js

### PR DESCRIPTION
**Pull request by gera2ld**
AMO requires third-party libraries to be at a published release instead of a copy of code ([ref](https://extensionworkshop.com/documentation/publish/third-party-library-usage/)).

So I'm going to drop the copy of `zip.js` in this repo and use an alternative from NPM. Finally I decide to use JSZip (which is more popular) instead of zip.js because of the friendly APIs and documentation.

### CODE REVIEW

**Comment by tophf** [comment]
Isn't it too big? ~120kB minified vs the old ~51kB. I also don't like that it adds 10ms to page load time because it loads in the main chunk whereas previously it was loaded dynamically.

**Comment by tophf** [comment]
>published release instead of a copy of code

How does this relate to VM?
Here's the published release: https://www.npmjs.com/package/@zip.js/zip.js

**Comment by tophf** [comment]
Also, JSZip is several times slower...

**Comment by tophf** [comment]
Ah, no, the speed is the same, the db was different.

**Comment by gera2ld** [comment]
JSZip uses pako under the hood so it's bigger in size.
The reason I don't want to stay with zip.js is the lack of documentation and comparation of different implementations... Anyway I'll try since it's much smaller and may be better in performance.

File: src/common/zip.js [hunk 1]
```diff
@@ -1,18 +1,8 @@
...
-    script.onload = () => resolve();
-    script.onerror = () => reject();
-    document.body.append(script);
-  });
-}
-
 const loadZip = memoize(async () => {
-  await loadScript('/public/lib/zip.js/zip.js');
-  return window.zip;
+  const zip = await import(/* webpackChunkName: 'zip' */ '@zip.js/zip.js/dist/zip');
```
Comment (tophf) [inline src/common/zip.js:4]: Neat! But it creates `vendors~zip.js` in the root directory so maybe add `...splitVendor('@zip'),` in plaid.conf.js?

Comment (gera2ld) [reply src/common/zip.js:4]: Done!

**Comment by tophf** [comment]
The library is 34kB bigger now... but the real problem is that it's much slower: the old zip.js exported 6MB of scripts into 1.4MB zip in 2 seconds, while the new one takes 6 seconds. In devtools Sources` panel I see a lot of workers are created for a fraction of a second and then destroyed. In the old zip.js there was just one worker.

**Comment by gera2ld** [comment]
I guess it's due to the consecutive `writer.add`s. Maybe I should replace them with concurrent ones.

File: src/options/views/tab-settings/vm-export.vue [hunk 2]
```diff
@@ -137,60 +114,57 @@ function normalizeFilename(name) {
...
+  });
+  files.push({
+    name: 'violentmonkey',
+    content: JSON.stringify(vm),
+  });
+  const zip = await loadZip();
+  const blobWriter = new zip.BlobWriter('application/zip');
+  const writer = new zip.ZipWriter(blobWriter);
+  await Promise.all(files.map(file => writer.add(file.name, new zip.TextReader(file.content), {
+    lastModDate: file.lastModDate,
```
Comment (tophf) [inline src/options/views/tab-settings/vm-export.vue:164]: Yeah it's fast now but I wonder why [the documentation](https://gildas-lormeau.github.io/zip.js/core-api.html#zip-writing) suggests `bufferedWrite: true`? The output is identical in both cases.

**Review by tophf — APPROVED** [review]

File: src/options/views/tab-settings/vm-export.vue (hunk 2 above)
Comment (gera2ld) [reply src/options/views/tab-settings/vm-export.vue:164]: The author has given a detailed clarification [here](https://github.com/gildas-lormeau/zip.js/issues/236).
I think it does not matter much in our case, but may make a difference when files are large.

**Pull request merged**